delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/Storages.scala

```diff
@@ -544,13 +573,15 @@ object Storages {
         config.amazon.as(StorageType.S3Storage) ++
         config.remoteDisk.as(StorageType.RemoteDiskStorage)
 
-    def validateAndReturnValue(id: Iri, fields: StorageFields): IO[StorageRejection, StorageValue] =
+    def validateAndReturnValue(id: Iri, fields: StorageFields): IO[StorageRejection, StorageValue[Encrypted]] =
       for {
-        value <- IO.fromOption(fields.toValue(config), InvalidStorageType(id, fields.tpe, allowedStorageTypes))
-        _     <- validatePermissions(fields)
-        _     <- access(value)
-        _     <- validateFileSize(id, fields.maxFileSize, value.maxFileSize)
-      } yield value
+        value     <- IO.fromOption(fields.toValue(config), InvalidStorageType(id, fields.tpe, allowedStorageTypes))
+        encrypted <- IO.fromEither(value.encrypt(config.encryption.crypto))
+        _         <- IO.fromEither(encrypted.decrypt(config.encryption.crypto))
```

Comment:
why are we attempting to decrypt it? is it to ensure it can be decrypted?

Reply:
Yes, since the next function will decrypt it unsafely, we have to check in the eval function if that can be done